Search issues

tonygoldcrest opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/peancored/vaadin-bugrap/3)
<!-- Reviewable:end -->

Commit:
Search issues and top menu

Commit:
Changed favicon and project title.

Commit:
Created issues filter component, that take filter object.
Refactored other components according new component.

Commit:
Partial refactor of filter. Just to restore commit on the new laptop

Commit:
Completed reports bar, that shows percents of non-resolved, closed and unassigned issues.

Saulis commented on src/components/select-project.html:
```diff
@@ -1,9 +1,3 @@
-
-<link rel="import" href="shared-styles.html">
-<link rel="import" href="../bower_components/vaadin-combo-box/vaadin-combo-box.html">
```
Even though `vaadin-combo-box` and `iron-ajax` are imported in a parent element, it's a good practice to always import dependencies inside the element that uses them. This is done to avoid breakage if there are changes at some point the parents' imports. Browser takes care of not importing anything multiple times so it's OK to have duplicates.

Commit:
Added button groups.
Styled Vaadin button.
Added versions selector.
Added versions dummy objects list.

manolo commented:
I merged the other PR, so I have changed the base branch of this one to master, you need to rebase with master and `push -f` to this branch

manolo commented on src/issues-list.html:
```diff
@@ -54,12 +52,36 @@
 
       static get properties () {
         return {
+          issueTitleFilter: {
```
Title does not sound good to me, probably subject sounds better.

manolo commented on src/components/top-menu.html:
```diff
@@ -0,0 +1,90 @@
+
+  <script>
+    class TopMenu extends Polymer.Element {
+      static get is() { return 'top-menu'; }
```
One line block is not in the syntax convention of the Team.
Not a blocker, but you must to learn and follow our syntax conventions.

manolo commented on src/components/bottom-filters.html:
```diff
@@ -0,0 +1,63 @@
+
+    <div class="row">
+      <span class="reports-for-text">Reports for:</span>
+      <vaadin-combo-box filtered-items="[[items]]" label="Project" item-label-path="title" on-filter-changed="_projectsFilter" selected-item="{{selectedProject}}">
```
why filtered-items and no items ?

manolo commented on index.html:
```diff
@@ -24,7 +24,7 @@
     <base href="/">
 
-    <link rel="icon" href="images/favicon.ico">
+    <link rel="icon" href="images/favicon.png">
```
Apart from the icon in the index.html, you need to modify the images referenced in the manifest.json file so as all icons are coherent in all devices supporting PWA

Commit:
Added functionality for radio button groups, added context menu.

Commit:
Refactored all is() fuctions according to multiple-line-blocks convention.

Commit:
Changed images links in manifest.json

Commit:
Fix folder structure and importing dependencies.

Commit:
Added checkboxes that filter issues by status.

manolo commented:
@Toshegg as a convention we use reviewable for all our components. I think it could be a good exercise if you add reviewable hook to your repo and become familiar with the tool. If you find some trouble how to setup the repo ask me.

Toshegg commented:
@manolo, I've added an integration.

manolo commented:
Nice, we can continue the review here

---

Review status: 0 of 18 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/toshegg/vaadin-bugrap/3#-:-KnZQKExZSN_2qGpftm9:btbg7ng)*
<!-- Sent from Reviewable.io -->

Saulis commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> nice work with the radio button. I left some minor comments, you can decide yourself if you implement them or not. Ready to merge from my part.

---

Review status: 0 of 18 files reviewed at latest revision, 1 unresolved discussion.

---

*[src/components/bottom-filters.html, line 132 at r3](https://reviewable.io:443/reviews/toshegg/vaadin-bugrap/3#-KnhdKuXoXJlK1Qx-HoY:-KnhdKuXoXJlK1Qx-HoZ:b11ele5) ([raw file](https://github.com/toshegg/vaadin-bugrap/blob/8a900e1ba8762d5ec44a8acb22b2002ef560da5a/src/components/bottom-filters.html#L132)):*
> ```HTML
>         super.ready();
> 
>         this.$.onlyMe.addEventListener('click', this._changeAssigneeFilter.bind(this, 'onlyMe'));
> ```

Just a comment: I would consider an approach where all these `click` listeners could be added declaratively to the buttons. 
For example, you could use the `id` of the button to detect what was clicked. So you could add `on-click="_changeAssigneeFilter"` and change the `_changeAssigneeFilter` to `switch(event.target.id)`  instead.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/toshegg/vaadin-bugrap/3#-:-KnhfK96_DcibJe0GZs5:bp18yee)*
<!-- Sent from Reviewable.io -->

Commit:
Refactor radio-button on-click event so now it works using event.target.id.

Toshegg commented:
Review status: 0 of 18 files reviewed at latest revision, 1 unresolved discussion.

---

*[src/components/bottom-filters.html, line 132 at r3](https://reviewable.io:443/reviews/toshegg/vaadin-bugrap/3#-KnhdKuXoXJlK1Qx-HoY:-KniNABXOp3X0LWL5aqA:bvecxtp) ([raw file](https://github.com/toshegg/vaadin-bugrap/blob/8a900e1ba8762d5ec44a8acb22b2002ef560da5a/src/components/bottom-filters.html#L132)):*
<details><summary><i>Previously, Saulis (Sauli Tähkäpää) wrote…</i></summary><blockquote>

Just a comment: I would consider an approach where all these `click` listeners could be added declaratively to the buttons. 
For example, you could use the `id` of the button to detect what was clicked. So you could add `on-click="_changeAssigneeFilter"` and change the `_changeAssigneeFilter` to `switch(event.target.id)`  instead.

</blockquote></details>

I've refactored that in next commit.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/toshegg/vaadin-bugrap/3)*
<!-- Sent from Reviewable.io -->

manolo commented:
Good job. <img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

---

Reviewed 4 of 14 files at r1, 3 of 8 files at r2, 13 of 14 files at r3, 1 of 1 files at r4.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/toshegg/vaadin-bugrap/3#-:-KnoyvAfqNZbm5cZHUqg:b-s5ml19)*
<!-- Sent from Reviewable.io -->

manolo commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[src/components/bottom-filters.html, line 132 at r3](https://reviewable.io:443/reviews/toshegg/vaadin-bugrap/3#-KnhdKuXoXJlK1Qx-HoY:-Knoz1GLpj1pE8A1TUlG:b-baqbcr) ([raw file](https://github.com/toshegg/vaadin-bugrap/blob/8a900e1ba8762d5ec44a8acb22b2002ef560da5a/src/components/bottom-filters.html#L132)):*
<details><summary><i>Previously, Toshegg wrote…</i></summary><blockquote>

I've refactored that in next commit.
</blockquote></details>

Needs ACK from Sauli in order to use reviewable for merging the PR. Otherwise you can use github interface and force the merging.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/toshegg/vaadin-bugrap/3)*
<!-- Sent from Reviewable.io -->
